mood_bot.py

```diff
@@ -0,0 +1,726 @@
+# mood_bot.py
+import asyncio
+import datetime
+import mood_error as me
+import os
+import re
+import shelve
+
+from discord.ext import commands
+from dotenv import load_dotenv
+
+
+load_dotenv()
+MOOD_TOKEN = os.getenv('DISCORD_TOKEN')
+
+mood_bot = commands.Bot(command_prefix='mood_')
+
+days_of_week = ('Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday')
+
+day_regex = re.compile(r'\b((mon|tues|wed(nes)?|thur(s)?|fri|sat(ur)?|sun)(day)?)|'
+                       r'(everyday|weekend(s)?|weekday(s)?)\b', re.I)
+
+day_dictionary = {'monday': 0, 'mon': 0, 'tuesday': 1, 'tues': 1, 'tue': 1, 'wednesday': 2, 'wednes': 2, 'wed': 2,
+                  'thursday': 3, 'thurs': 3, 'thur': 3, 'friday': 4, 'fri': 4, 'saturday': 5, 'satur': 5, 'sat': 5,
+                  'sunday': 6, 'sun': 6}
+
+""" These are the preset choices for mood check-ins. """
+mood_choices = ('good', 'bad', 'okay', 'meh', 'happy', 'sad', 'angry', 'upset', 'depressed', 'hungry',
+                'tired', 'bored', 'irritated', 'stressed', 'frustrated', 'ashamed', 'guilty', 'worried',
+                'lonely', 'prideful', 'confused', 'disgusted', 'surprised', 'disappointed')
+
+"""Power is for switching the auto_checker ON/OFF"""
+auto_checker_power_switch = 'OFF'
+
+""" Gates are for the mood check execution; to make sure that the mood check is done in the correct order. """
+date_gate = 'Open'
+type_gate = 'Closed'
+rating_gate = 'Closed'
+why_gate = 'Closed'
+stage_array = ['date', 'type', 'rating', 'explanation']

```

Comment:
I'm a bit concerned that with this setup, we use one gate over all of our users, or worse, we do not persist the gate between commands.  We could do either storing this in mongo or more reasonably with a smaller number of users we could use a dictionary with user ids b/c this might make more sense to cache then to move to permanent storage

Reply:
We might want to have @Angel1412kaitou think about what happens if a user aborts the mood check in halfway through the process

Reply:
Currently, saving to an external file, not sure how to cache. As for the aborting process, I do have a reset function that goes off automatically if a new auto-check is started before another is finished. This function deletes the data from the incomplete mood check.

Reply:
I like Nick's idea. Sorry I've been MIA (and plan on being MIA until this semester ends) but you guys are doing great